runtime/service_integ_test.go

```diff
@@ -316,3 +329,99 @@ func TestMultipleVMs_Isolated(t *testing.T) {
+
+	client, err := containerd.New(containerdSockPath, containerd.WithDefaultRuntime(firecrackerRuntime))
+	require.NoError(t, err, "unable to create client to containerd service at %s, is containerd running?", containerdSockPath)
+	defer client.Close()
+
+	image, err := client.Pull(ctx, debianDockerImage, containerd.WithPullUnpack, containerd.WithPullSnapshotter(naiveSnapshotterName))
+	require.NoError(t, err, "failed to pull image %s, is the the %s snapshotter running?", debianDockerImage, naiveSnapshotterName)
+
+	rootfsBytes, err := ioutil.ReadFile(defaultVMRootfsPath)
+	require.NoError(t, err, "failed to read rootfs file")
+
+	tapName := fmt.Sprintf("tap%d", vmID)
+	err = createTapDevice(ctx, tapName)
+	require.NoError(t, err, "failed to create tap device for vm %d", vmID)
+
+	// TODO once Noah's immutable rootfs change is merged, we can use that as our rootfs for all VMs instead of copying
+	// one per-VM
+	rootfsPath := fmt.Sprintf("%s.%d", defaultVMRootfsPath, vmID)
+	err = ioutil.WriteFile(rootfsPath, rootfsBytes, 0600)
+	require.NoError(t, err, "failed to copy vm rootfs to %s", rootfsPath)
+
+	containerName := fmt.Sprintf("%s-%d", t.Name(), time.Now().UnixNano())
+	snapshotName := fmt.Sprintf("%s-snapshot", containerName)
+
+	fcClient := fccontrol.NewFirecrackerClient(client.Conn())
+	_, err = fcClient.CreateVM(ctx, &proto.CreateVMRequest{
+		VMID: strconv.Itoa(vmID),
+		RootDrive: &proto.FirecrackerDrive{
+			PathOnHost:   rootfsPath,
+			IsReadOnly:   false,
+			IsRootDevice: true,
+		},
+		NetworkInterfaces: []*proto.FirecrackerNetworkInterface{
+			{
+				HostDevName: tapName,
+				MacAddress:  vmIDtoMacAddr(uint(vmID)),
+				AllowMMDS:   true,
+			},
+		},
+		ContainerCount: 5,
+	})
+
+	newContainer, err := client.NewContainer(ctx,
+		containerName,
+		containerd.WithSnapshotter(naiveSnapshotterName),
+		containerd.WithNewSnapshot(snapshotName, image),
+		containerd.WithNewSpec(
+			oci.WithProcessArgs("lsblk"),
+			firecrackeroci.WithVMID(strconv.Itoa(vmID)),
+		),
+	)
+	require.NoError(t, err, "failed to create container %s", containerName)
+
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	newTask, err := newContainer.NewTask(ctx,
+		cio.NewCreator(cio.WithStreams(nil, bufio.NewWriter(&stdout), bufio.NewWriter(&stderr))))
+	require.NoError(t, err, "failed to create task for container %s", containerName)
+
+	err = newTask.Start(ctx)
+	require.NoError(t, err, "failed to start task for container %s", containerName)
+
+	exitCh, err := newTask.Wait(ctx)
+	require.NoError(t, err, "failed to wait on task for container %s", containerName)
+
+	const containerID = 0
+
+	select {
+	case exitStatus := <-exitCh:
+		// if there was anything on stderr, print it to assist debugging
+		stderrOutput := stderr.String()
+		if len(stderrOutput) != 0 {
+			fmt.Printf("stderr output from vm %d, container %d: %s", vmID, containerID, stderrOutput)
+		}
+
+		const expectedOutput = `NAME MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
```

Comment:
I don't have a better suggestion for you right now, but I'm not really wild about depending on both the output of a specific version of `lsblk` and the fact that `lsblk` is in the default Debian image.